fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineShortcuts.kt

```diff
@@ -0,0 +1,273 @@
+        )
+    }
+}
+
+@Suppress("LongParameterList", "LongMethod")
+@Composable
+private fun SearchItem(
+    engine: SearchEngine,
+    name: String,
+    isEnabled: Boolean,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+) {
+    val isMenuExpanded: MutableState<Boolean> = remember { mutableStateOf(false) }
+
+    Row(
+        modifier = Modifier
+            .defaultMinSize(minHeight = 56.dp)
+            .padding(start = 4.dp),
+    ) {
+        Checkbox(
+            modifier = Modifier.align(Alignment.CenterVertically),
+            checked = isEnabled,
+            onCheckedChange = { onCheckboxClicked.invoke(engine, it) },
+            colors = CheckboxDefaults.colors(
+                checkedColor = FirefoxTheme.colors.formSelected,
+            ),
+        )
+
+        Spacer(modifier = Modifier.width(20.dp))
+
+        Image(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .size(24.dp),
+            bitmap = engine.icon.asImageBitmap(),
+            contentDescription = stringResource(
+                id = R.string.search_engine_icon_content_description,
+                engine.name,
+            ),
+        )
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Text(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .padding(top = 8.dp, bottom = 8.dp)
+                .weight(1f),
+            text = name,
+            style = FirefoxTheme.typography.subtitle1,
+            color = FirefoxTheme.colors.textPrimary,
+        )
+
+        if (engine.type == SearchEngine.Type.CUSTOM) {
+            Box(
+                modifier = Modifier
+                    .align(Alignment.CenterVertically),
+            ) {
+                IconButton(
+                    onClick = {
+                        isMenuExpanded.value = true
+                    },
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_menu),
+                        contentDescription = stringResource(id = R.string.content_description_menu),
+                        tint = FirefoxTheme.colors.iconPrimary,
+                    )
+
+                    SearchEngineDropdownMenu(
+                        engine = engine,
+                        isMenuExpanded = isMenuExpanded,
+                        onEditEngineClicked = onEditEngineClicked,
+                        onDeleteEngineClicked = onDeleteEngineClicked,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun SearchEngineDropdownMenu(
+    engine: SearchEngine,
+    isMenuExpanded: MutableState<Boolean>,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+) {
+    DropdownMenu(
```

Comment:
we have a `ContextualMenu` composable component, which should save you some time here.

Reply:
implemented